### PR TITLE
feat: Identity provider setup, ucauth pkg refactor.

### PR DESCRIPTION
**Pull request by shuofan**
#### What this PR does / why we need it:
Identity provider setup, ucauth pkg refactor.

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMjk3LDEyODYsMTI4MCwxMjUwLDEzMjFdLCJhc3NpZ25lZSI6WyIxMDAxMDczIl19&id=316029&iterationID=1297&pId=319600&type=TASK)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |  Identity provider setup, ucauth pkg refactor.            |
| 🇨🇳 中文    |    用户身份 provider          |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: cmd/core-services/bootstrap.yaml [hunk 1]
```diff
@@ -48,4 +48,6 @@ audit:
 
 erda.core.token:
 
-erda.core.org:
\ No newline at end of file
+erda.core.org:
+
+erda.core.user.identity:
```
Comment (sfwn) [inline cmd/core-services/bootstrap.yaml:53]: use `erda.core.user` directly.

File: internal/core/user/identity/provider.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	"github.com/jinzhu/gorm"
+
+	"github.com/erda-project/erda-infra/base/logs"
+	"github.com/erda-project/erda-infra/base/servicehub"
+	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/pkg/discover"
+	"github.com/erda-project/erda/pkg/http/httpclient"
+)
+
+type config struct {
```
Comment (sfwn) [inline internal/core/user/identity/provider.go:29]: split into `kratos` and `terminusuc`

File: internal/core/user/identity/provider.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	UCClientSecret       string `default:"secret" file:"UC_CLIENT_SECRET"`
+	OryKratosPrivateAddr string `default:"kratos-admin" file:"ORY_KRATOS_ADMIN_ADDR"`
+}
+
+type Interface interface {
+	FindUsers(ids []string) ([]User, error)
+	FindUsersByKey(key string) ([]User, error)
+	GetUser(userID string) (*User, error)
+	GetUsers(IDs []string, needDesensitize bool) (map[string]apistructs.UserInfo, error)
+	UserMigration(req OryKratosCreateIdentitiyRequest) (string, error)
```
Comment (sfwn) [inline internal/core/user/identity/provider.go:41]: move these last two inteface to kratos privately.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Comment by sfwn** [comment]
the whole dir structure should be:

- internal/core/user/
  - kratos/
  - terminusuc/
    - uc-adaptor/
  - provider.go
  - ...

File: cmd/core-services/bootstrap.yaml (hunk 1 above)
Comment (shuofan) [reply cmd/core-services/bootstrap.yaml]: done

File: internal/core/user/identity/provider.go (hunk 2 above)
Comment (shuofan) [reply internal/core/user/identity/provider.go]: done

File: internal/core/user/identity/provider.go (hunk 3 above)
Comment (shuofan) [reply internal/core/user/identity/provider.go]: done

File: internal/core/user/uc-adaptor/ucclient/ucclient.go [hunk 4]
```diff
@@ -26,11 +26,11 @@ import (
 	"github.com/sirupsen/logrus"
 
 	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/internal/core/user/uc"
```
Comment (sfwn) [inline internal/core/user/uc-adaptor/ucclient/ucclient.go:29]: move uc-adaptor under uc dir.

Comment (sfwn) [reply internal/core/user/uc-adaptor/ucclient/ucclient.go:29]: internal/core/user/
- impl/kratos/
- impl/uc/
  - impl/uc/uc-adaptor/

handler/ is strange.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/core/user/uc-adaptor/ucclient/ucclient.go (hunk 4 above)
Comment (shuofan) [reply internal/core/user/uc-adaptor/ucclient/ucclient.go]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**